src/gentropy/dataset/dataset.py

```diff
@@ -172,7 +191,7 @@ def validate_schema(self: Dataset) -> None:
         Raises:
             SchemaValidationError: If the DataFrame schema does not match the expected schema
         """
-        expected_schema = self._schema
+        expected_schema = self.schema
```

Comment:
Isn't `self.schema` inconsistent with line 40? Where the schema is specified as `_schema: StructType | None = None`.